pkg/controllers/dynakube/otelc/secret/reconciler.go

```diff
@@ -0,0 +1,132 @@
+package secret
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/otelc/consts"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/conditions"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/hasher"
+	k8slabels "github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/labels"
+	k8ssecret "github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/secret"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Reconciler struct {
+	client    client.Client
+	apiReader client.Reader
+	dk        *dynakube.DynaKube
+}
+
+type ReconcilerBuilder func(client client.Client, apiReader client.Reader, dk *dynakube.DynaKube) *Reconciler
+
+func NewReconciler(client client.Client, apiReader client.Reader, dk *dynakube.DynaKube) *Reconciler {
+	return &Reconciler{
+		client:    client,
+		dk:        dk,
+		apiReader: apiReader,
+	}
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context) error {
+	if r.dk.TelemetryService().IsEnabled() {
+		return r.ensureTelemetryServiceApiCredentialsSecret(ctx)
+	}
+
+	return r.removeTelemetryServiceApiCredentialsSecret(ctx)
+}
+
+func (r *Reconciler) ensureTelemetryServiceApiCredentialsSecret(ctx context.Context) error {
+	query := k8ssecret.Query(r.client, r.apiReader, log)
+	_, err := query.Get(ctx, types.NamespacedName{Name: consts.TelemetryApiCredentialsSecretName, Namespace: r.dk.Namespace})
+
+	if err != nil && k8serrors.IsNotFound(err) {
```

Comment:
What should we do if there happens to be an other error than `IsNotFound`?

Or is this just an unrealistic scenario?

Reply:
read line 47

Reply:
? 😅  (I only see that we do nothing)

Reply:
@andriisoldatenko 
line 47: 
```go
_, err := query.Get(ctx, types.NamespacedName{Name: consts.TelemetryApiCredentialsSecretName, Namespace: r.dk.Namespace})
```

I don't feel this answeres the question 😬  (or am I missing something)

Reply:
what about kspm?
```
func ensureKSPMSecret(ctx context.Context, client client.Client, apiReader client.Reader, dk *dynakube.DynaKube) error {
	query := k8ssecret.Query(client, apiReader, log)
	_, err := query.Get(ctx, types.NamespacedName{Name: dk.KSPM().GetTokenSecretName(), Namespace: dk.Namespace})

	if err != nil && k8serrors.IsNotFound(err) {
```
not saying it's correct, just it's consistent. I can add check and return error.

Reply:
I will look at the `kspm` one and add it there

Reply:
let me fix it here.

Reply:
fix for the KSPM one: https://github.com/Dynatrace/dynatrace-operator/pull/4561

Reply:
fixed in 27c11f85e

Reply:
fixed in fc68f1e67